<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>References | Bacterial filamentation: a bet for survival in stressful environments</title>
<meta name="author" content="Vélez Santiago Jesús">
<meta name="description" content="Ackermann, Martin. “A Functional Perspective on Phenotypic Heterogeneity in Microorganisms.” Nature Reviews Microbiology 13, no. 8 (July 6, 2015): 497–508. https://doi.org/10.1038/nrmicro3491. ...">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="References | Bacterial filamentation: a bet for survival in stressful environments">
<meta property="og:type" content="book">
<meta property="og:url" content="https://jvelezmagic.github.io/undergraduate_research_project/index.html/references.html">
<meta property="og:description" content="Ackermann, Martin. “A Functional Perspective on Phenotypic Heterogeneity in Microorganisms.” Nature Reviews Microbiology 13, no. 8 (July 6, 2015): 497–508. https://doi.org/10.1038/nrmicro3491. ...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="References | Bacterial filamentation: a bet for survival in stressful environments">
<meta name="twitter:description" content="Ackermann, Martin. “A Functional Perspective on Phenotypic Heterogeneity in Microorganisms.” Nature Reviews Microbiology 13, no. 8 (July 6, 2015): 497–508. https://doi.org/10.1038/nrmicro3491. ...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="css/style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Bacterial filamentation: a bet for survival in stressful environments</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome</a></li>
<li><a class="" href="introduction.html">Introduction</a></li>
<li><a class="" href="image-processing.html"><span class="header-section-number">1</span> Image processing</a></li>
<li><a class="" href="experiment-analysis.html"><span class="header-section-number">2</span> Experiment analysis</a></li>
<li><a class="" href="model-analysis.html"><span class="header-section-number">3</span> Models to the rescue; filamentation abstraction</a></li>
<li><a class="" href="general-discussion.html"><span class="header-section-number">4</span> Discussion</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="code-availability.html"><span class="header-section-number">A</span> Code availability</a></li>
<li><a class="" href="software-tools.html"><span class="header-section-number">B</span> Software tools</a></li>
<li><a class="" href="software-usage.html"><span class="header-section-number">C</span> Software usage</a></li>
<li><a class="" href="colophon.html"><span class="header-section-number">D</span> Colophon</a></li>
<li><a class="active" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/jvelezmagic/undergraduate_research_project">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="references" class="section level1 unnumbered">
<h1>References<a class="anchor" aria-label="anchor" href="#references"><i class="fas fa-link"></i></a>
</h1>
<!-- # References {-} -->

<div id="refs" class="references">
<div id="ref-ackermann2015">
<p>Ackermann, Martin. “A Functional Perspective on Phenotypic Heterogeneity in Microorganisms.” <em>Nature Reviews Microbiology</em> 13, no. 8 (July 6, 2015): 497–508. <a href="https://doi.org/10.1038/nrmicro3491">https://doi.org/10.1038/nrmicro3491</a>.</p>
</div>
<div id="ref-R-rmarkdown">
<p>Allaire, JJ, Yihui Xie, Jonathan McPherson, Javier Luraschi, Kevin Ushey, Aron Atkins, Hadley Wickham, Joe Cheng, Winston Chang, and Richard Iannone. <em>Rmarkdown: Dynamic Documents for R</em>, 2021. <a href="https://CRAN.R-project.org/package=rmarkdown">https://CRAN.R-project.org/package=rmarkdown</a>.</p>
</div>
<div id="ref-altman1992">
<p>Altman, N. S. “An Introduction to Kernel and Nearest-Neighbor Nonparametric Regression.” <em>The American Statistician</em> 46, no. 3 (August 1992): 175–85. <a href="https://doi.org/10.1080/00031305.1992.10475879">https://doi.org/10.1080/00031305.1992.10475879</a>.</p>
</div>
<div id="ref-andersson2005">
<p>Andersson, Dan I. “The Ways in Which Bacteria Resist Antibiotics.” <em>International Journal of Risk and Safety in Medicine</em> 17, nos. 3-4 (2005): 111–16.</p>
</div>
<div id="ref-andrews2001">
<p>Andrews, Jennifer M. “Determination of Minimum Inhibitory Concentrations.” <em>Journal of Antimicrobial Chemotherapy</em> 48, no. suppl_1 (July 1, 2001): 5–16. <a href="https://doi.org/10.1093/jac/48.suppl_1.5">https://doi.org/10.1093/jac/48.suppl_1.5</a>.</p>
</div>
<div id="ref-andrewsDeterminationMinimumInhibitory2002">
<p>Andrews, J. M. “Determination of Minimum Inhibitory Concentrations.” <em>Journal of Antimicrobial Chemotherapy</em> 49, no. 6 (June 2002): 1049–9. <a href="https://doi.org/10.1093/jac/dkf083">https://doi.org/10.1093/jac/dkf083</a>.</p>
</div>
<div id="ref-balaban2004a">
<p>Balaban, Nathalie Q., Jack Merrin, Remy Chait, Lukasz Kowalik, and Stanislas Leibler. “Bacterial Persistence as a Phenotypic Switch.” <em>Science</em> 305, no. 5690 (September 10, 2004): 1622–5. <a href="https://doi.org/10.1126/science.1099390">https://doi.org/10.1126/science.1099390</a>.</p>
</div>
<div id="ref-Julia-2017">
<p>Bezanson, Jeff, Alan Edelman, Stefan Karpinski, and Viral B Shah. “Julia: A Fresh Approach to Numerical Computing.” <em>SIAM Review</em> 59, no. 1 (2017): 65–98. <a href="https://doi.org/10.1137/141000671">https://doi.org/10.1137/141000671</a>.</p>
</div>
<div id="ref-bosEmergenceAntibioticResistance2015">
<p>Bos, Julia, Qiucen Zhang, Saurabh Vyawahare, Elizabeth Rogers, Susan M. Rosenberg, and Robert H. Austin. “Emergence of Antibiotic Resistance from Multinucleated Bacterial Filaments.” <em>Proceedings of the National Academy of Sciences</em> 112, no. 1 (January 2015): 178–83. <a href="https://doi.org/10.1073/pnas.1420702111">https://doi.org/10.1073/pnas.1420702111</a>.</p>
</div>
<div id="ref-10.5555/1386553">
<p>Brinkmann, Ron. <em>The Art and Science of Digital Compositing, Second Edition: Techniques for Visual Effects, Animation and Motion Graphics (the Morgan Kaufmann Series in ... Morgan Kaufmann Series in Computer Graphics)</em>. 2nd ed. San Francisco, CA, USA: Morgan Kaufmann Publishers Inc., 2008.</p>
</div>
<div id="ref-bruggeman">
<p>Bruggeman, Frank J., Jorrit J. Hornberg, Fred C. Boogerd, and Hans V. Westerhoff. “Introduction to Systems Biology,” 1–19. Birkhäuser Basel, 2007. <a href="https://doi.org/10.1007/978-3-7643-7439-6_1">https://doi.org/10.1007/978-3-7643-7439-6_1</a>.</p>
</div>
<div id="ref-Caicedo2017">
<p>Caicedo, Juan C, Sam Cooper, Florian Heigwer, Scott Warchal, Peng Qiu, Csaba Molnar, Aliaksei S Vasilevich, et al. “Data-Analysis Strategies for Image-Based Cell Profiling.” <em>Nature Methods</em> 14, no. 9 (September 2017): 849–63. <a href="https://doi.org/10.1038/nmeth.4397">https://doi.org/10.1038/nmeth.4397</a>.</p>
</div>
<div id="ref-camposConstantSizeExtension2014">
<p>Campos, Manuel, Ivan V. Surovtsev, Setsu Kato, Ahmad Paintdakhi, Bruno Beltran, Sarah E. Ebmeier, and Christine Jacobs-Wagner. “A Constant Size Extension Drives Bacterial Cell Size Homeostasis.” <em>Cell</em> 159, no. 6 (December 2014): 1433–46. <a href="https://doi.org/10.1016/j.cell.2014.11.022">https://doi.org/10.1016/j.cell.2014.11.022</a>.</p>
</div>
<div id="ref-cayron2020">
<p>Cayron, Julien, Annick Dedieu, and Christian Lesterlin. “Bacterial Filament Division Dynamics Allows Rapid Post-Stress Cell Proliferation,” March 16, 2020. <a href="http://dx.doi.org/10.1101/2020.03.16.993345">http://dx.doi.org/10.1101/2020.03.16.993345</a>.</p>
</div>
<div id="ref-convery2019">
<p>Convery, Neil, and Nikolaj Gadegaard. “30 Years of Microfluidics.” <em>Micro and Nano Engineering</em> 2 (March 2019): 76–91. <a href="https://doi.org/10.1016/j.mne.2019.01.003">https://doi.org/10.1016/j.mne.2019.01.003</a>.</p>
</div>
<div id="ref-datseris2020">
<p>Datseris, George, Jonas Isensee, Sebastian Pech, and Tamás Gál. “DrWatson: The Perfect Sidekick for Your Scientific Inquiries.” <em>Journal of Open Source Software</em> 5, no. 54 (October 29, 2020): 2673. <a href="https://doi.org/10.21105/joss.02673">https://doi.org/10.21105/joss.02673</a>.</p>
</div>
<div id="ref-dever1991">
<p>Dever, L. A., and T. S. Dermody. “Mechanisms of Bacterial Resistance to Antibiotics.” <em>Archives of Internal Medicine</em> 151, no. 5 (May 1991): 886–95.</p>
</div>
<div id="ref-EditorialBoard2014">
<p>“Editorial Board.” <em>Journal of Global Antimicrobial Resistance</em> 2, no. 2 (June 2014): ii. <a href="https://doi.org/10.1016/S2213-7165(14)00044-7">https://doi.org/10.1016/S2213-7165(14)00044-7</a>.</p>
</div>
<div id="ref-elowitz2002">
<p>Elowitz, Michael B., Arnold J. Levine, Eric D. Siggia, and Peter S. Swain. “Stochastic Gene Expression in a Single Cell.” <em>Science</em> 297, no. 5584 (August 16, 2002): 1183–6. <a href="https://doi.org/10.1126/science.1070919">https://doi.org/10.1126/science.1070919</a>.</p>
</div>
<div id="ref-R-janitor">
<p>Firke, Sam. <em>Janitor: Simple Tools for Examining and Cleaning Dirty Data</em>, 2021. <a href="https://github.com/sfirke/janitor">https://github.com/sfirke/janitor</a>.</p>
</div>
<div id="ref-friedmanPreciseTemporalModulation2005">
<p>Friedman, Nir, Shuki Vardi, Michal Ronen, Uri Alon, and Joel Stavans. “Precise Temporal Modulation in the Response of the SOS DNA Repair Network in Individual Bacteria.” Edited by B’en’edicte Michel. <em>PLoS Biology</em> 3, no. 7 (June 2005): e238. <a href="https://doi.org/10.1371/journal.pbio.0030238">https://doi.org/10.1371/journal.pbio.0030238</a>.</p>
</div>
<div id="ref-shapely2007">
<p>Gillies, Sean, and others. “Shapely: Manipulation and Analysis of Geometric Objects.” toblerity.org, n.d. <a href="https://github.com/Toblerity/Shapely">https://github.com/Toblerity/Shapely</a>.</p>
</div>
<div id="ref-hagberg2008exploring">
<p>Hagberg, Aric, Pieter Swart, and Daniel S Chult. “Exploring Network Structure, Dynamics, and Function Using Networkx.” Los Alamos National Lab.(LANL), Los Alamos, NM (United States), 2008.</p>
</div>
<div id="ref-2020NumPy-Array">
<p>Harris, Charles R., K. Jarrod Millman, Stéfan J van der Walt, Ralf Gommers, Pauli Virtanen, David Cournapeau, Eric Wieser, et al. “Array Programming with NumPy.” <em>Nature</em> 585 (2020): 357–62. <a href="https://doi.org/10.1038/s41586-020-2649-2">https://doi.org/10.1038/s41586-020-2649-2</a>.</p>
</div>
<div id="ref-harrisRelativeRatesSurface2016">
<p>Harris, Leigh K., and Julie A. Theriot. “Relative Rates of Surface and Volume Synthesis Set Bacterial Cell Size.” <em>Cell</em> 165, no. 6 (June 2016): 1479–92. <a href="https://doi.org/10.1016/j.cell.2016.05.045">https://doi.org/10.1016/j.cell.2016.05.045</a>.</p>
</div>
<div id="ref-heinrich2015">
<p>Heinrich, Kristina, David J. Leslie, and Kristina Jonas. “Modulation of Bacterial Proliferation as a Survival Strategy,” 127–71. Elsevier, 2015. <a href="https://doi.org/10.1016/bs.aambs.2015.02.004">https://doi.org/10.1016/bs.aambs.2015.02.004</a>.</p>
</div>
<div id="ref-R-fs">
<p>Hester, Jim, Hadley Wickham, and Gábor Csárdi. <em>Fs: Cross-Platform File System Operations Based on Libuv</em>, 2021. <a href="https://CRAN.R-project.org/package=fs">https://CRAN.R-project.org/package=fs</a>.</p>
</div>
<div id="ref-hotelling1936">
<p>Hotelling, Harold. “Relations Between Two Sets of Variates.” <em>Biometrika</em> 28, no. 3/4 (December 1936): 321. <a href="https://doi.org/10.2307/2333955">https://doi.org/10.2307/2333955</a>.</p>
</div>
<div id="ref-hunter2007matplotlib">
<p>Hunter, John D. “Matplotlib: A 2D Graphics Environment.” <em>Computing in Science &amp; Engineering</em> 9, no. 3 (2007): 90–95.</p>
</div>
<div id="ref-jaimes-lizcano2014">
<p>Jaimes-Lizcano, Yuly A., Dayton D. Hunn, and Kyriakos D. Papadopoulos. “Filamentous Escherichia Coli Cells Swimming in Tapered Microcapillaries.” <em>Research in Microbiology</em> 165, no. 3 (April 2014): 166–74. <a href="https://doi.org/10.1016/j.resmic.2014.01.007">https://doi.org/10.1016/j.resmic.2014.01.007</a>.</p>
</div>
<div id="ref-Joseleau-Petit1999">
<p>Joseleau-Petit, Daniel̀e, Daniel Vinella, and Richard D’Ari. “Metabolic Alarms and Cell Division in Escherichia Coli.” <em>Journal of Bacteriology</em> 181, no. 1 (January 1999): 9–14. <a href="https://doi.org/10.1128/jb.181.1.9-14.1999">https://doi.org/10.1128/jb.181.1.9-14.1999</a>.</p>
</div>
<div id="ref-justiceMorphologicalPlasticityBacterial2008">
<p>Justice, Sheryl S., David A. Hunstad, Lynette Cegelski, and Scott J. Hultgren. “Morphological Plasticity as a Bacterial Survival Strategy.” <em>Nature Reviews Microbiology</em> 6, no. 2 (February 2008): 162–68. <a href="https://doi.org/10.1038/nrmicro1820">https://doi.org/10.1038/nrmicro1820</a>.</p>
</div>
<div id="ref-justiceFilamentationEscherichiaColi2006">
<p>Justice, S. S., D. A. Hunstad, P. C. Seed, and S. J. Hultgren. “Filamentation by Escherichia Coli Subverts Innate Defenses During Urinary Tract Infection.” <em>Proceedings of the National Academy of Sciences</em> 103, no. 52 (December 2006): 19884–9. <a href="https://doi.org/10.1073/pnas.0606329104">https://doi.org/10.1073/pnas.0606329104</a>.</p>
</div>
<div id="ref-R-ggpubr">
<p>Kassambara, Alboukadel. <em>Ggpubr: Ggplot2 Based Publication Ready Plots</em>, 2020. <a href="https://rpkgs.datanovia.com/ggpubr/">https://rpkgs.datanovia.com/ggpubr/</a>.</p>
</div>
<div id="ref-R-ggdist">
<p>Kay, Matthew. <em>Ggdist: Visualizations of Distributions and Uncertainty</em>, 2021. <a href="https://CRAN.R-project.org/package=ggdist">https://CRAN.R-project.org/package=ggdist</a>.</p>
</div>
<div id="ref-kroemer2008">
<p>Kroemer, G, L Galluzzi, P Vandenabeele, J Abrams, E S Alnemri, E H Baehrecke, M V Blagosklonny, et al. “Classification of Cell Death: Recommendations of the Nomenclature Committee on Cell Death 2009.” <em>Cell Death &amp; Differentiation</em> 16, no. 1 (October 10, 2008): 3–11. <a href="https://doi.org/10.1038/cdd.2008.150">https://doi.org/10.1038/cdd.2008.150</a>.</p>
</div>
<div id="ref-R-embed">
<p>Kuhn, Max. <em>Embed: Extra Recipes for Encoding Predictors</em>, 2021. <a href="https://CRAN.R-project.org/package=embed">https://CRAN.R-project.org/package=embed</a>.</p>
</div>
<div id="ref-R-tidymodels">
<p>Kuhn, Max, and Hadley Wickham. <em>Tidymodels: Easily Install and Load the Tidymodels Packages</em>, 2021. <a href="https://CRAN.R-project.org/package=tidymodels">https://CRAN.R-project.org/package=tidymodels</a>.</p>
</div>
<div id="ref-lambertBacterialResistanceAntibiotics2005">
<p>Lambert, P. “Bacterial Resistance to Antibiotics: Modified Target Sites.” <em>Advanced Drug Delivery Reviews</em> 57, no. 10 (July 2005): 1471–85. <a href="https://doi.org/10.1016/j.addr.2005.04.003">https://doi.org/10.1016/j.addr.2005.04.003</a>.</p>
</div>
<div id="ref-mcinnes2018umap">
<p>McInnes, Leland, John Healy, and James Melville. “Umap: Uniform Manifold Approximation and Projection for Dimension Reduction.” <em>arXiv Preprint arXiv:1802.03426</em>, 2018.</p>
</div>
<div id="ref-mckinney2010data">
<p>McKinney, Wes, and others. “Data Structures for Statistical Computing in Python.” In <em>Proceedings of the 9th Python in Science Conference</em>, 445:51–56. Austin, TX, 2010.</p>
</div>
<div id="ref-meldrumFacultyOpinionsRecommendation2005">
<p>Meldrum, Deirdre. “Faculty Opinions Recommendation of Bacterial Persistence as a Phenotypic Switch.” January 2005. <a href="https://doi.org/10.3410/f.1021664.264823">https://doi.org/10.3410/f.1021664.264823</a>.</p>
</div>
<div id="ref-millerSOSResponseInduction2004">
<p>Miller, C. “SOS Response Induction by -Lactams and Bacterial Defense Against Antibiotic Lethality.” <em>Science</em> 305, no. 5690 (September 2004): 1629–31. <a href="https://doi.org/10.1126/science.1101630">https://doi.org/10.1126/science.1101630</a>.</p>
</div>
<div id="ref-million-weaverMechanismsPlasmidSegregation2014">
<p>Million-Weaver, Samuel, and Manel Camps. “Mechanisms of Plasmid Segregation: Have Multicopy Plasmids Been Overlooked?” <em>Plasmid</em> 75 (September 2014): 27–36. <a href="https://doi.org/10.1016/j.plasmid.2014.07.002">https://doi.org/10.1016/j.plasmid.2014.07.002</a>.</p>
</div>
<div id="ref-moger-reischer2019">
<p>Moger-Reischer, Roy Z., and Jay T. Lennon. “Microbial Ageing and Longevity.” <em>Nature Reviews Microbiology</em> 17, no. 11 (September 18, 2019): 679–90. <a href="https://doi.org/10.1038/s41579-019-0253-y">https://doi.org/10.1038/s41579-019-0253-y</a>.</p>
</div>
<div id="ref-mückl2018">
<p>Mückl, Andrea, Matthaeus Schwarz-Schilling, Katrin Fischer, and Friedrich C. Simmel. “Filamentation and Restoration of Normal Growth in Escherichia Coli Using a Combined Crispri sgRNA/Antisense Rna Approach.” Edited by Mark Isalan. <em>PLOS ONE</em> 13, no. 9 (September 11, 2018): e0198058. <a href="https://doi.org/10.1371/journal.pone.0198058">https://doi.org/10.1371/journal.pone.0198058</a>.</p>
</div>
<div id="ref-R-here">
<p>Müller, Kirill. <em>Here: A Simpler Way to Find Your Files</em>, 2020. <a href="https://CRAN.R-project.org/package=here">https://CRAN.R-project.org/package=here</a>.</p>
</div>
<div id="ref-neidhardtBacterialGrowthConstant1999">
<p>Neidhardt, Frederick C. “Bacterial Growth: Constant Obsession with <em>dN</em><em>/Dt</em>.” <em>Journal of Bacteriology</em> 181, no. 24 (December 1999): 7405–8. <a href="https://doi.org/10.1128/JB.181.24.7405-7408.1999">https://doi.org/10.1128/JB.181.24.7405-7408.1999</a>.</p>
</div>
<div id="ref-pearson1901">
<p>Pearson, Karl. “LIII. On Lines and Planes of Closest Fit to Systems of Points in Space.” <em>The London, Edinburgh, and Dublin Philosophical Magazine and Journal of Science</em> 2, no. 11 (November 1901): 559–72. <a href="https://doi.org/10.1080/14786440109462720">https://doi.org/10.1080/14786440109462720</a>.</p>
</div>
<div id="ref-R-patchwork">
<p>Pedersen, Thomas Lin. <em>Patchwork: The Composer of Plots</em>, 2020. <a href="https://CRAN.R-project.org/package=patchwork">https://CRAN.R-project.org/package=patchwork</a>.</p>
</div>
<div id="ref-perez2007ipython">
<p>Pérez, Fernando, and Brian E Granger. “IPython: A System for Interactive Scientific Computing.” <em>Computing in Science &amp; Engineering</em> 9, no. 3 (2007).</p>
</div>
<div id="ref-rackauckas2017adaptive">
<p>Rackauckas, Christopher, and Qing Nie. “Adaptive Methods for Stochastic Differential Equations via Natural Embeddings and Rejection Sampling with Memory.” <em>Discrete and Continuous Dynamical Systems. Series B</em> 22, no. 7 (2017): 2731.</p>
</div>
<div id="ref-rackauckas2017differentialequations">
<p>———. “Differentialequations.jl–a Performant and Feature-Rich Ecosystem for Solving Differential Equations in Julia.” <em>Journal of Open Research Software</em> 5, no. 1 (2017).</p>
</div>
<div id="ref-rackauckas_stability-optimized_2018">
<p>———. “Stability-Optimized High Order Methods and Stiffness Detection for Pathwise Stiff Stochastic Differential Equations.” <em>arXiv:1804.04344 [Math]</em>, 2018. <a href="http://arxiv.org/abs/1804.04344">http://arxiv.org/abs/1804.04344</a>.</p>
</div>
<div id="ref-R-base">
<p>R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing, 2021. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.</p>
</div>
<div id="ref-R-tidytext">
<p>Robinson, David, and Julia Silge. <em>Tidytext: Text Mining Using Dplyr, Ggplot2, and Other Tidy Tools</em>, 2021. <a href="https://github.com/juliasilge/tidytext">https://github.com/juliasilge/tidytext</a>.</p>
</div>
<div id="ref-rocklin2015dask">
<p>Rocklin, Matthew. “Dask: Parallel Computation with Blocked Algorithms and Task Scheduling.” In <em>Proceedings of the 14th Python in Science Conference</em>. 130-136. Citeseer, 2015.</p>
</div>
<div id="ref-roostalu2008">
<p>Roostalu, Johanna, Arvi Jõers, Hannes Luidalepp, Niilo Kaldalu, and Tanel Tenson. “Cell Division in Escherichia Colicultures Monitored at Single Cell Resolution.” <em>BMC Microbiology</em> 8, no. 1 (April 23, 2008). <a href="https://doi.org/10.1186/1471-2180-8-68">https://doi.org/10.1186/1471-2180-8-68</a>.</p>
</div>
<div id="ref-sanmillan2016">
<p>San Millan, Alvaro, Jose Antonio Escudero, Danna R. Gifford, Didier Mazel, and R. Craig MacLean. “Multicopy Plasmids Potentiate the Evolution of Antibiotic Resistance in Bacteria.” <em>Nature Ecology &amp; Evolution</em> 1, no. 1 (November 7, 2016). <a href="https://doi.org/10.1038/s41559-016-0010">https://doi.org/10.1038/s41559-016-0010</a>.</p>
</div>
<div id="ref-satoOuterMembranePermeability1991">
<p>Sato, Kazuhito, and Taiji Nakae. “Outer Membrane Permeability of <em>Acinetobacter</em> <em>Calcoaceticus</em> and Its Implication in Antibiotic Resistance.” <em>Journal of Antimicrobial Chemotherapy</em> 28, no. 1 (1991): 35–45. <a href="https://doi.org/10.1093/jac/28.1.35">https://doi.org/10.1093/jac/28.1.35</a>.</p>
</div>
<div id="ref-schaechterDependencyMediumTemperature1958">
<p>Schaechter, M., O. MaalOe, and N. O. Kjeldgaard. “Dependency on Medium and Temperature of Cell Size and Chemical Composition During Balanced Growth of Salmonella Typhimurium.” <em>Journal of General Microbiology</em> 19, no. 3 (December 1958): 592–606. <a href="https://doi.org/10.1099/00221287-19-3-592">https://doi.org/10.1099/00221287-19-3-592</a>.</p>
</div>
<div id="ref-schaechterGrowthCellNuclear1962">
<p>Schaechter, M., J. P. Williamson, J. R. Hood, and A. L. Koch. “Growth, Cell and Nuclear Divisions in Some Bacteria.” <em>Journal of General Microbiology</em> 29, no. 3 (November 1962): 421–34. <a href="https://doi.org/10.1099/00221287-29-3-421">https://doi.org/10.1099/00221287-29-3-421</a>.</p>
</div>
<div id="ref-schermelleh2019">
<p>Schermelleh, Lothar, Alexia Ferrand, Thomas Huser, Christian Eggeling, Markus Sauer, Oliver Biehlmaier, and Gregor P. C. Drummen. “Super-Resolution Microscopy Demystified.” <em>Nature Cell Biology</em> 21, no. 1 (January 2019): 72–84. <a href="https://doi.org/10.1038/s41556-018-0251-8">https://doi.org/10.1038/s41556-018-0251-8</a>.</p>
</div>
<div id="ref-R-GGally">
<p>Schloerke, Barret, Di Cook, Joseph Larmarange, Francois Briatte, Moritz Marbach, Edwin Thoen, Amos Elberg, and Jason Crowley. <em>GGally: Extension to Ggplot2</em>, 2021. <a href="https://CRAN.R-project.org/package=GGally">https://CRAN.R-project.org/package=GGally</a>.</p>
</div>
<div id="ref-schneider2012">
<p>Schneider, Caroline A, Wayne S Rasband, and Kevin W Eliceiri. “NIH Image to Imagej: 25 Years of Image Analysis.” <em>Nature Methods</em> 9, no. 7 (June 28, 2012): 671–75. <a href="https://doi.org/10.1038/nmeth.2089">https://doi.org/10.1038/nmeth.2089</a>.</p>
</div>
<div id="ref-R-plotly">
<p>Sievert, Carson, Chris Parmer, Toby Hocking, Scott Chamberlain, Karthik Ram, Marianne Corvellec, and Pedro Despouy. <em>Plotly: Create Interactive Web Graphics via Plotly.js</em>, 2021. <a href="https://CRAN.R-project.org/package=plotly">https://CRAN.R-project.org/package=plotly</a>.</p>
</div>
<div id="ref-sliusarenkoHighthroughputSubpixelPrecision2011">
<p>Sliusarenko, Oleksii, Jennifer Heinritz, Thierry Emonet, and Christine Jacobs-Wagner. “High-Throughput, Subpixel Precision Analysis of Bacterial Morphogenesis and Intracellular Spatio-Temporal Dynamics: Quantitative Analysis of Spatio-Temporal Dynamics.” <em>Molecular Microbiology</em> 80, no. 3 (May 2011): 612–27. <a href="https://doi.org/10.1111/j.1365-2958.2011.07579.x">https://doi.org/10.1111/j.1365-2958.2011.07579.x</a>.</p>
</div>
<div id="ref-Smith2018">
<p>Smith, Kevin, Filippo Piccinini, Tamas Balassa, Krisztian Koos, Tivadar Danka, Hossein Azizpour, and Peter Horvath. “Phenotypic Image Analysis Software Tools for Exploring and Understanding Big Image Data from Cell-Based Assays.” <em>Cell Systems</em> 6, no. 6 (June 2018): 636–53. <a href="https://doi.org/10.1016/j.cels.2018.06.001">https://doi.org/10.1016/j.cels.2018.06.001</a>.</p>
</div>
<div id="ref-sofroniew2021a">
<p>Sofroniew, Nicholas, Talley Lambert, Kira Evans, Juan Nunez-Iglesias, Grzegorz Bokota, Gonzalo Peña-Castellanos, Philip Winston, et al. <em>Napari/Napari: 0.4.12rc2</em>. Zenodo, 2021. <a href="https://doi.org/10.5281/ZENODO.3555620">https://doi.org/10.5281/ZENODO.3555620</a>.</p>
</div>
<div id="ref-Specht2017">
<p>Specht, Elizabeth A., Esther Braselmann, and Amy E. Palmer. “A Critical and Comparative Review of Fluorescent Tools for Live-Cell Imaging.” <em>Annual Review of Physiology</em> 79, no. 1 (February 10, 2017): 93–117. <a href="https://doi.org/10.1146/annurev-physiol-022516-034055">https://doi.org/10.1146/annurev-physiol-022516-034055</a>.</p>
</div>
<div id="ref-taheri-araghiCellSizeControlHomeostasis2017">
<p>Taheri-Araghi, Sattar, Serena Bradde, John T. Sauls, Norbert S. Hill, Petra Anne Levin, Johan Paulsson, Massimo Vergassola, and Suckjoon Jun. “Cell-Size Control and Homeostasis in Bacteria.” <em>Current Biology</em> 27, no. 9 (May 2017): 1392. <a href="https://doi.org/10.1016/j.cub.2017.04.028">https://doi.org/10.1016/j.cub.2017.04.028</a>.</p>
</div>
<div id="ref-trevors2012">
<p>Trevors, J.T. “Can Dead Bacterial Cells Be Defined and Are Genes Expressed After Cell Death?” <em>Journal of Microbiological Methods</em> 90, no. 1 (July 2012): 25–28. <a href="https://doi.org/10.1016/j.mimet.2012.04.004">https://doi.org/10.1016/j.mimet.2012.04.004</a>.</p>
</div>
<div id="ref-turing1950">
<p>TURING, A. M. “I.COMPUTING Machinery and Intelligence.” <em>Mind</em> LIX, no. 236 (October 1, 1950): 433–60. <a href="https://doi.org/10.1093/mind/lix.236.433">https://doi.org/10.1093/mind/lix.236.433</a>.</p>
</div>
<div id="ref-ursellRapidPreciseQuantification2017">
<p>Ursell, Tristan, Timothy K. Lee, Daisuke Shiomi, Handuo Shi, Carolina Tropini, Russell D. Monds, Alexandre Colavin, et al. “Rapid, Precise Quantification of Bacterial Cellular Dimensions Across a Genomic-Scale Knockout Library.” <em>BMC Biology</em> 15, no. 1 (December 2017): 17. <a href="https://doi.org/10.1186/s12915-017-0348-8">https://doi.org/10.1186/s12915-017-0348-8</a>.</p>
</div>
<div id="ref-R-renv">
<p>Ushey, Kevin. <em>Renv: Project Environments</em>, 2021. <a href="https://rstudio.github.io/renv/">https://rstudio.github.io/renv/</a>.</p>
</div>
<div id="ref-vanderwalt2014">
<p>van der Walt, Stéfan, Johannes L. Schönberger, Juan Nunez-Iglesias, François Boulogne, Joshua D. Warner, Neil Yager, Emmanuelle Gouillart, and Tony Yu. “Scikit-Image: Image Processing in Python.” <em>PeerJ</em> 2 (June 19, 2014): e453. <a href="https://doi.org/10.7717/peerj.453">https://doi.org/10.7717/peerj.453</a>.</p>
</div>
<div id="ref-van1995python">
<p>Van Rossum, Guido. <em>The Python Library Reference, Release 3.8.2</em>. Python Software Foundation, 2020.</p>
</div>
<div id="ref-10.5555/1593511">
<p>Van Rossum, Guido, and Fred L. Drake. <em>Python 3 Reference Manual</em>. Scotts Valley, CA: CreateSpace, 2009.</p>
</div>
<div id="ref-vanvalen2016">
<p>Van Valen, David A., Takamasa Kudo, Keara M. Lane, Derek N. Macklin, Nicolas T. Quach, Mialy M. DeFelice, Inbal Maayan, Yu Tanouchi, Euan A. Ashley, and Markus W. Covert. “Deep Learning Automates the Quantitative Analysis of Individual Cells in Live-Cell Imaging Experiments.” Edited by Martin Meier-Schellersheim. <em>PLOS Computational Biology</em> 12, no. 11 (November 4, 2016): e1005177. <a href="https://doi.org/10.1371/journal.pcbi.1005177">https://doi.org/10.1371/journal.pcbi.1005177</a>.</p>
</div>
<div id="ref-verschaffel2002">
<p>Verschaffel, Lieven, Brian Greer, and Erik de Corte. “Everyday Knowledge and Mathematical Modeling of School Word Problems,” 257–76. Springer Netherlands, 2002. <a href="https://doi.org/10.1007/978-94-017-3194-2_16">https://doi.org/10.1007/978-94-017-3194-2_16</a>.</p>
</div>
<div id="ref-wang2009">
<p>Wang, Jue D., and Petra A. Levin. “Metabolism, Cell Growth and the Bacterial Cell Cycle.” <em>Nature Reviews Microbiology</em> 7, no. 11 (October 6, 2009): 822–27. <a href="https://doi.org/10.1038/nrmicro2202">https://doi.org/10.1038/nrmicro2202</a>.</p>
</div>
<div id="ref-Wang2014">
<p>Wang, Ying, Hong Wu, Xiaoran Jiang, and Guo-Qiang Chen. “Engineering Escherichia Coli for Enhanced Production of Poly(3-Hydroxybutyrate-Co-4-Hydroxybutyrate) in Larger Cellular Space.” <em>Metabolic Engineering</em> 25 (September 2014): 183–93. <a href="https://doi.org/10.1016/j.ymben.2014.07.010">https://doi.org/10.1016/j.ymben.2014.07.010</a>.</p>
</div>
<div id="ref-Wang2014a">
<p>Wang, Ying, Jin Yin, and Guo-Qiang Chen. “Polyhydroxyalkanoates, Challenges and Opportunities.” <em>Current Opinion in Biotechnology</em> 30 (December 2014): 59–65. <a href="https://doi.org/10.1016/j.copbio.2014.06.001">https://doi.org/10.1016/j.copbio.2014.06.001</a>.</p>
</div>
<div id="ref-webberImportanceEffluxPumps2003">
<p>Webber, M. A. “The Importance of Efflux Pumps in Bacterial Antibiotic Resistance.” <em>Journal of Antimicrobial Chemotherapy</em> 51, no. 1 (January 2003): 9–11. <a href="https://doi.org/10.1093/jac/dkg050">https://doi.org/10.1093/jac/dkg050</a>.</p>
</div>
<div id="ref-white2021">
<p>White, John Myles, Bogumił Kamiński, Powerdistribution, Milan Bouchet-Valat, Sean Garborg, Jacob Quinn, Simon Kornblith, et al. <em>JuliaData/Dataframes.jl: V1.2.2</em>. Zenodo, 2021. <a href="https://doi.org/10.5281/ZENODO.3376177">https://doi.org/10.5281/ZENODO.3376177</a>.</p>
</div>
<div id="ref-R-stringr">
<p>Wickham, Hadley. <em>Stringr: Simple, Consistent Wrappers for Common String Operations</em>, 2019. <a href="https://CRAN.R-project.org/package=stringr">https://CRAN.R-project.org/package=stringr</a>.</p>
</div>
<div id="ref-R-tidyverse">
<p>———. <em>Tidyverse: Easily Install and Load the Tidyverse</em>, 2021. <a href="https://CRAN.R-project.org/package=tidyverse">https://CRAN.R-project.org/package=tidyverse</a>.</p>
</div>
<div id="ref-R-sessioninfo">
<p>Wickham, Hadley, Winston Chang, Robert Flight, Kirill Müller, and Jim Hester. <em>Sessioninfo: R Session Information</em>, 2021. <a href="https://CRAN.R-project.org/package=sessioninfo">https://CRAN.R-project.org/package=sessioninfo</a>.</p>
</div>
<div id="ref-wrightBacterialResistanceAntibiotics2005">
<p>Wright, G. “Bacterial Resistance to Antibiotics: Enzymatic Degradation and Modification.” <em>Advanced Drug Delivery Reviews</em> 57, no. 10 (July 2005): 1451–70. <a href="https://doi.org/10.1016/j.addr.2005.04.002">https://doi.org/10.1016/j.addr.2005.04.002</a>.</p>
</div>
<div id="ref-R-bookdown">
<p>Xie, Yihui. <em>Bookdown: Authoring Books and Technical Documents with R Markdown</em>, 2021. <a href="https://CRAN.R-project.org/package=bookdown">https://CRAN.R-project.org/package=bookdown</a>.</p>
</div>
<div id="ref-R-knitr">
<p>———. <em>Knitr: A General-Purpose Package for Dynamic Report Generation in R</em>, 2021. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.</p>
</div>
<div id="ref-yin2012">
<p>Yin, Huabing, and Damian Marshall. “Microfluidics for Single Cell Analysis.” <em>Current Opinion in Biotechnology</em> 23, no. 1 (February 2012): 110–19. <a href="https://doi.org/10.1016/j.copbio.2011.11.002">https://doi.org/10.1016/j.copbio.2011.11.002</a>.</p>
</div>
</div>
</div>
























































































  <div class="chapter-nav">
<div class="prev"><a href="colophon.html"><span class="header-section-number">D</span> Colophon</a></div>
<div class="empty"></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#references">References</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/jvelezmagic/undergraduate_research_project/blob/main/99-references.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/jvelezmagic/undergraduate_research_project/edit/main/99-references.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Bacterial filamentation: a bet for survival in stressful environments</strong>" was written by Vélez Santiago Jesús. It was last built on 2022-01-03.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
